diff -up azmq-1.0.3/test/socket/main.cpp.omv~ azmq-1.0.3/test/socket/main.cpp
--- azmq-1.0.3/test/socket/main.cpp.omv~	2024-09-29 17:10:09.211968184 +0200
+++ azmq-1.0.3/test/socket/main.cpp	2024-09-29 17:10:18.592070059 +0200
@@ -13,6 +13,7 @@
 #include <boost/algorithm/string/split.hpp>
 #include <boost/algorithm/string/classification.hpp>
 #include <boost/asio/buffer.hpp>
+#include <boost/current_function.hpp>
 
 #include <array>
 #include <thread>
diff -up azmq-1.0.3/test/socket_ops/main.cpp.omv~ azmq-1.0.3/test/socket_ops/main.cpp
--- azmq-1.0.3/test/socket_ops/main.cpp.omv~	2024-09-29 17:10:45.552362907 +0200
+++ azmq-1.0.3/test/socket_ops/main.cpp	2024-09-29 17:10:54.462459699 +0200
@@ -2,6 +2,7 @@
 #include <azmq/detail/socket_ops.hpp>
 
 #include <boost/asio/buffer.hpp>
+#include <boost/current_function.hpp>
 
 #include <array>
 #include <chrono>
